msq: add durable storage info (#14035)
apache/druid: docs/multi-stage-query/reference.md

## Durable Storage

Using durable storage with your SQL-based ingestions can improve their reliability by writing intermediate files to a storage location temporarily. 

To prevent durable storage from getting filled up with temporary files in case the tasks fail to clean them up, a periodic
cleaner can be scheduled to clean the directories corresponding to which there isn't a controller task running. It utilizes
the storage connector to work upon the durable storage. The durable storage location should only be utilized to store the output
for cluster's MSQ tasks. If the location contains other files or directories, then they will get cleaned up as well.

### Enable durable storage

To enable durable storage, you need to set the following common service properties:

```
druid.msq.intermediate.storage.enable=true
druid.msq.intermediate.storage.type=s3
druid.msq.intermediate.storage.bucket=YOUR_BUCKET
druid.msq.intermediate.storage.prefix=YOUR_PREFIX
druid.msq.intermediate.storage.tempDir=/path/to/your/temp/dir
```

> For information about these settings and others related to durable storage, see [Durable storage configurations](#durable-storage-configurations).

Reviewer: ```suggestion
For detailed information about the settings related to durable storage, see [Durable storage configurations](#durable-storage-configurations).
```

| `druid.msq.intermediate.storage.chunkSize` | n/a | Optional. Defines the size of each chunk to temporarily store in `druid.msq.intermediate.storage.tempDir`. The chunk size must be between 5 MiB and 5 GiB. Druid computes the chunk size automatically if no value is provided.| 
|`druid.msq.intermediate.storage.enable` | true | Required. Whether to enable durable storage for the cluster.|
| `druid.msq.intermediate.storage.maxTriesOnTransientErrors` | 10 | Optional. Defines the max number times to attempt S3 API calls to avoid failures due to transient errors. | 
|`druid.msq.intermediate.storage.type` | `s3` if your deep storage is S3 | Required. The type of storage to use. You can either set this to `local` or `s3`.  |
|`druid.msq.intermediate.storage.prefix` | n/a | S3 prefix to store intermediate stage results. Provide a unique value for the prefix. Don't share the same prefix between clusters. If the location  includes other files or directories, then they will get cleaned up as well.  |
| `druid.msq.intermediate.storage.tempDir`|  | Required. Directory path on the local disk to temporarily store intermediate stage results.  |

> In addition to the common service properties, there are certain properties that you configure on the Overlord specifically:

Reviewer: ```suggestion
In addition to the common service properties, there are certain properties that you configure on the Overlord specifically to clean up intermediate files:
```
